app/scripts/metamask-controller.js

```diff
@@ -3514,7 +3515,41 @@ export default class MetamaskController extends EventEmitter {
    * @param {object} [req] - The original request, containing the origin.
    */
   async newUnapprovedTransaction(txParams, req) {
-    return await this.txController.newUnapprovedTransaction(txParams, req);
+    // Options are passed explicitly as an additional security measure
+    // to ensure approval is not disabled
+    const { result } = await this.txController.addTransaction(txParams, {
+      actionId: req.id,
+      method: req.method,
+      origin: req.origin,
+      // This is the default behaviour but specified here for clarity
+      requireApproval: true,
+    });
+
+    return await result;
```

Comment:
I'd argue it would only be simpler if you wanted to wait for the result.

This is the central benefit of this new method signature, as the caller can now choose whether they are only concerned with adding a transaction but don't care about approval or submission (the previous `addUnapprovedTransaction`), or they can wait for approval and submission (the previous `newUnapprovedTransaction`).

This is also the existing return type in the core `TransactionController` meaning the central method signature is now much more aligned going forward.

There's an argument that awaiting the result could be conditional inside the method, but that would diverge from the core `TransactionController`, plus greatly limit the clarity of the return value as what you're awaiting would be dynamic and so less predictable, plus you'd have an undefined value for the transaction hash if you chose not to await the result for example.